File: PennyPincher.API/PennyPincher/Data/Expense.cs
namespace PennyPincher.Data;

public enum CategoryTypes
{
    Undefined = 0,
    None = 1,
    Living = 1,
    Utilities = 2, 
    Entertainment = 3,
    Shopping = 4,
    Takeout = 5
}

public class Expense
{
    public int Id { get; set; }

    public string Name { get; set; } = string.Empty;
    
    public CategoryTypes Category { get; set; }
    
    public double Price { get; set; } 
}

File: PennyPincherUI/src/app/models/expense.ts
export class ExpenseDto {
	constructor(
		public id: number,
		public name: string,
		public category: CATEGORY_TYPES,
		public price: number,
	) {}
}
export class ExpenseForUpdateDto {
	constructor(
		public name: string,
		public price: number,
		public category: CATEGORY_TYPES,
	) {}
}

export enum CATEGORY_TYPES {
	None = 0,
	Living = 1,
	Utilities = 2,
	Entertainment = 3,
	Shopping = 4,
	Takeout = 5,
}
